### PR TITLE
Make compatible with new keras Callback model

### PR DESCRIPTION
**Pull request by ali-tny**
see https://keras.io/guides/writing_your_own_callbacks/ for guide - methods have been split to provide a different one for train/eval/predict time.

This change translates `on_batch_end` to `on_train_batch_end` rather than also for validation batches, so batch history only includes train batch loss - not sure if you wanna include the validation batches in the history here for plotting? Would just need to set `on_test_batch_end` to the same as `on_train_batch_end`.

### CODE REVIEW

**Review by ig248 — CHANGES_REQUESTED** [review]
Could you indicate what versions of keras the old vs new api would be compatible to? Also, maybe update the `Callback` docstring - tbh the only reason the base class is here is to avoid importing it from `keras`, hence avoiding `keras` as a dependency. However, if there are API differences, we could specify a keras version range in the setup.py?

**Comment by ali-tny** [comment]
I've updated to be backwards compatible (much [like tensorflow.keras does](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/callbacks.py)). I considered just adding the keras dependency and importing the Callback base class, but to install tf.keras it needs all of tensorflow, and it seems kinda heavyweight now that its compatible across versions.

I also added a test - it requires adding matplotlib & ipython as dev dependencies though so it can import `kerasplothistory.callbacks`
